gammapy/image/tests/test_lists.py

```diff
@@ -0,0 +1,22 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+from gammapy.datasets import FermiGalacticCenter
+import numpy as np
+
+
+def test_to_cube():
+    sky_cube_original = FermiGalacticCenter.diffuse_model()
+    image_list = sky_cube_original.to_image_list()
+    assert np.array_equal(image_list.skymaps[0].data, sky_cube_original.data[0])
+    sky_cube_restored = image_list.to_cube()
+
+    for key in set(sky_cube_original.__dict__.keys() + sky_cube_restored.__dict__.keys()):
```

Comment:
I think this code that's looping over keys in the data `__dict__` objects is too complicated.
By reading the test it's not clear to me what the keys will be and what exactly happens.

I would prefer to have simple assert statements on the individual data members of the cube
(see https://gammapy.readthedocs.io/en/latest/api/gammapy.cube.SkyCube.html), checking that it's the same.
I.e. roughtly:
```
# assert cube1.data == cube2.data
# assert cube1.wcs == cube2.wcs
# assert cube1.energy == cube2.energy
```
(I'm not sure where you can really just check with `==` or where a more complex assert statement is needed.)